### PR TITLE
KLS-741 - Display summary of filters applied on event listing page

### PR DESCRIPTION
**Pull request by mforner13**
This PR adds a 'results summary' at the top of the event listing page. This summary changes depending on whether the user is looking at all events, current bookings or past bookings.

**To test locally**
- Go to `export-academy/events`
- Apply the filters to see the filter summary appear
- If you don't have any events appearing - make sure to publish some future events locally [here](http://greatcms.trade.great:8020/admin/events/)

### **All events**
Screenshot 2023-06-12 at 15 45 52

**All events no results**
Screenshot 2023-06-12 at 15 45 17

### **Current bookings**
Screenshot 2023-06-12 at 15 46 57
**Current bookings no results**
Screenshot 2023-06-12 at 15 47 12

### **Past bookings**
Screenshot 2023-06-12 at 15 48 55

Screenshot 2023-06-12 at 15 48 38

### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/KLS-741
- [x] Jira ticket has the correct status.
- [x] A clear/description pull request messaged added.

### Reviewing help

- [x] Explains how to test locally, including how to set up appropriate data
- [x] Includes screenshot(s) - ideally before and after, but at least after

### Housekeeping

- [x] Frontend assets have been re-compiled.

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: export_academy/templatetags/event_list_buttons.py [hunk 1]
```diff
@@ -51,3 +51,24 @@ def get_filters(list_of_filters, user):
...
+    """
+    applied_filters = []
+    for filter_type, filter_choices in filter_form.data.lists():
+        if filter_type != 'booking_period':
+            for filter_choice in filter_choices:
+                for value, text in filter_form.fields[filter_type].choices:
+                    if value == filter_choice and value != filters.EventFilter.ALL:
+                        applied_filters.append(text)
+    return applied_filters
+
```
Comment (mforner13) [inline export_academy/templatetags/event_list_buttons.py:69]: Any suggestions for simplifying this would be greatly appreciated 🙏

Comment (bsabhilash) [reply export_academy/templatetags/event_list_buttons.py:69]: Few ideas I can think of, but none better than original :)


```
applied_filters = [
    text
    for filter_type, filter_choices in filter_form.data.lists()
    if filter_type != 'booking_period'
    for filter_choice in filter_choices
    for value, text in filter_form.fields[filter_type].choices
    if value == filter_choice and value != filters.EventFilter.ALL
]
return applied_filters
```

or modular

```
def get_applied_filters(filter_form):
    applied_filters = []

    for filter_type, filter_choices in filter_form.data.lists():
        if filter_type != 'booking_period':
            for filter_choice in filter_choices:
                filter_text = get_filter_text(filter_form, filter_type, filter_choice)
                if filter_text is not None:
                    applied_filters.append(filter_text)

    return applied_filters


def get_filter_text(filter_form, filter_type, filter_choice):
    for value, text in filter_form.fields[filter_type].choices:
        if value == filter_choice and value != filters.EventFilter.ALL:
            return text

    return None
```

File: export_academy/templates/export_academy/includes/filter_summary.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{% load content_tags %}
+{% load event_list_buttons %}
+<div class="govuk-!-margin-left-3 ea-filter-summary">
+    {% all_booking_periods_showing request.GET as all_booking_periods_showing %}
+    {% get_applied_filters filter.form as applied_filters %}
+    {% if all_booking_periods_showing and applied_filters|length > 0 %}
+        <p class="govuk-!-margin-top-3">Showing {{ page_obj.paginator.count|make_bold }} of {{ filter.queryset.count|make_bold}} events, filtered by:</p>
+        <p class="govuk-!-font-weight-bold">{{ applied_filters|join:", "|make_bold }}</p>
+        <a class="link govuk-!-margin-bottom-0 govuk-link--no-visited-state" href={% url 'export_academy:upcoming-events' %}>Clear filters</a>
```
Comment (lewis-coulson-dit) [inline export_academy/templates/export_academy/includes/filter_summary.html:9]: `link` to `govuk-link`

File: export_academy/templatetags/event_list_buttons.py (hunk 1 above)
Comment (lewis-coulson-dit) [reply export_academy/templatetags/event_list_buttons.py:69]: I'd just put lines 64-67 in their own function so your brain only every sees 1/2 `for` for `if` in one go :)

**Pull request merged**